cpp/memilio/math/integrator.h

```diff
@@ -127,19 +130,30 @@ class OdeIntegrator
             dt_copy = dt;
 
             results.add_time_point();
-            step_okay &= m_core->step(f, results[i], t, dt, results[i + 1]);
-            results.get_last_time() = t;
+            bool step_okay = m_core->step(f, results[i], t, dt, results[i + 1]);
 
             // if dt has been changed (even slighly) by step, register the current m_core as adaptive
             m_is_adaptive |= !floating_point_equal(dt, dt_copy);
+
+            // catch case where dt_min of an adaptive stepper is larger than tmax - t
+            if (t > tmax) {
+                log_info("Forcing last step size.");
+                // reset time and step size
+                dt = dt_copy;
+                t  = tmax - dt_copy;
+                // recalculate the erroneous last step and overwrite its results
+                step_okay = m_core->step(f, results[i], t, dt, results[i + 1], true);
+            }
+            is_step_sizing_okay &= step_okay;
```

Comment:
I think it would be enough to just use one bool.

Reply:
With only one bool and a forced last step it is difficult to report what the state before the forced step was. I assume that if we discard the step that went beyond tmax, we also want to discard its "step_okay" value. And if that step was not okay (i.e. false), the &= will overwrite the previous state, making it unrecoverable.

Do you agree with this assumption?